src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,97 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final int VALUE_INDEX = 1;
+    private static final int WORD_INDEX = 0;
+    private static final int SUPPLY_INDEX = 0;
+    private static final int BUY_INDEX = 1;
+    private static final int RESULT_INDEX = 2;
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT_VALUE = "result";
+    private static final String ZERO_VAR = "0";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String statistic = readText(fromFileName);
+        writeText(toFileName, statistic);
+    }
+
+    private void writeText(String toFileName, String statistic) {
+        try (BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(toFileName))) {
+            bufferedWriter.write(statistic);
+        } catch (IOException e) {
+            throw new RuntimeException("can't write to file");
+        }
+    }
+
+    private String readText(String fromFileName) {
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFileName))) {
+            return getResultText(bufferedReader);
```

Comment:
This method should be responsible only for reading the data and collecting it into a single place, but you are effectively calling 3 methods inside this try block inside each other. What we should do is minimize the code and logic amount inside the try block, and separate the report generation functionality from the file reading one